wkcuber/image_readers.py

```diff
@@ -63,7 +77,13 @@ def to_target_datatype(data: np.ndarray, target_dtype: np.dtype) -> np.ndarray:
 
 
 class Dm3ImageReader(ImageReader):
-    def read_array(self, file_name: str, dtype: np.dtype, z_slice: int) -> np.ndarray:
+    def read_array(
+        self,
+        file_name: str,
+        dtype: np.dtype,
+        z_slice: int,
+        channel_index: Optional[int],
```

Comment:
Currently, our DM3 and DM4 reader only read single channel, so this should be already checked. Since the only channel one can select is `0`, this check should guard against all cases:
`assert (
            channel_index < num_channels
        ), "Selected channel is not present in the input files"`

Reply:
Perfect